src/corona_hakab_model/generation/matrix_generator.py

```diff
@@ -62,63 +59,50 @@ def import_connection_data(import_file_path: str) -> "ConnectionData":
 
 
 class MatrixData:
-    __slots__ = ("matrix_type", "matrix", "depth")
+    def __getstate__(self):
```

Comment:
Why not implement _setstate_ ?

Reply:
you don't need it. you need `getstate` for the pickling. I'll add it as a comment